prediction_market_agent_tooling/markets/seer/price_manager.py

```
@@ -0,0 +1,119 @@
+        # in both sides hence we need current probabilities for both outcomes.
+        if price_yes and price_no:
+            # If other outcome`s price is None, we set it to 0.
+            total_price = sum(
+                price if price is not None else 0.0 for price in price_data.values()
+            )
+            price_yes = price_yes / total_price
+            return Probability(price_yes)
+        else:
+            return None
+
+    @lru_cache(typed=True)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)